modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php

```diff
@@ -328,14 +328,15 @@ public function enqueue_scripts() {
 	 * Enqueue WP.com block editor common styles.
 	 */
 	public function enqueue_styles() {
-		// Enqueue only for the block editor in WP Admin.
-		global $pagenow;
-		if ( is_admin() && ! in_array( $pagenow, array( 'post.php', 'post-new.php' ), true ) ) {
+		// On the front-end, add Justify styles inline when Gutenberg styles are also loaded.
+		if ( ! is_admin() ) {
+			wp_add_inline_style( 'wp-block-library', '.has-text-align-justify{text-align:justify;}' );
 			return;
 		}
 
-		// Enqueue on the front-end only if justified blocks are present.
-		if ( ! is_admin() && ! $this->has_justified_block() ) {
+		// Enqueue only for the block editor in WP Admin.
+		global $pagenow;
+		if ( ! in_array( $pagenow, array( 'post.php', 'post-new.php' ), true ) ) {
```

Comment:
I think this is not needed since Core makes sure the `enqueue_block_assets` hook is triggered only on the block editor or on the front-end:

https://core.trac.wordpress.org/browser/trunk/src/wp-includes/script-loader.php#L2836

So we could remove this check and enqueue the `wpcom-block-editor/common.css` style straight away.